### PR TITLE
[MXNET-806] Report error when shape mismatch in "where" operator

### PR DESCRIPTION
**Pull request by apeforest**
## Description ##
This is a bugfix for issue #11084 
```mx.sym.where(cond, x, y)``` still functions, given shape-mismatched cond, whereas ```mx.nd.where(cond, x, y)``` will throw exception. We should make sure the behavior is the same.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness
test_operator.py:test_where():test_invalid_shape()
- [ ] Code is well-documented: 
- This change is consistent with the current API documentation. Therefore, no additional documentation should be needed.
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] where operator with mismatched shape should throw exception

## Comments ##

### CODE REVIEW

File: tests/python/unittest/test_operator.py [hunk 1]
```diff
@@ -4442,6 +4442,18 @@ def test_where_numeric_gradient(shape, same_shape):
             condition_np, x_np, y_np = get_forward_inputs_condition_vector(shape)
         check_numeric_gradient(where_sym, [condition_np, x_np, y_np], grad_nodes=['x', 'y'])
 
+    def test_invalid_shape():
+        condition = mx.sym.Variable('condition')
+        x = mx.sym.Variable('x')
+        y = mx.sym.Variable('y')
+        where_sym = mx.sym.where(condition, x, y)
```
Comment (anirudh2290) [inline tests/python/unittest/test_operator.py]: can you add similar tests for mx.nd.where

Comment (apeforest) [reply tests/python/unittest/test_operator.py]: @anirudh2290 Added test for mx.nd.where.

**Comment by apeforest** [comment]
@anirudh2290  Can we merge this PR if no other reviewers comments?

File: src/operator/tensor/control_flow_op.h [hunk 2]
```diff
@@ -188,7 +188,7 @@ inline bool WhereOpShape(const nnvm::NodeAttrs& attrs,
     SHAPE_ASSIGN_CHECK(*in_attrs, 0, tshape);
     return true;
   } else if ((*in_attrs)[0].ndim() == 1) {
-    return (*in_attrs)[0].Size() == static_cast<size_t>(tshape[0]);
+    CHECK_EQ((*in_attrs)[0].Size(), static_cast<size_t>(tshape[0]));
```
Comment (zheng-da) [inline src/operator/tensor/control_flow_op.h]: this is wrong. what is the problem of the original code? If the first dimension doesn't match, infer shape fails, so it should return false.

**Pull request merged**